lib/theme_check/position_helper.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+# Note: Everything is 0-indexed here.
+
+module ThemeCheck
+  module PositionHelper
```

Comment:
Why split in two between Position and PositionHelper? How do you determine where to add a new position related method?

Maybe just a comment would make it clear.

Reply:
Also, `bounded` is used in `Position` which breaks the decoupling.

Reply:
PositionHelper is used in the LSP in a different context (to convert row/col info from the client into index. 2D -> 1D)

Moreover:

* PositionHelper is concerned about travelling from 1D to 2D and vice versa
* Position is concerned about stuff with a needle, a haystack and a line_number

Maybe I'm naming the second one poorly? Signature:

```
Position.new(needle, content, line_number)
```